phpstan.neon.dist

```diff
@@ -0,0 +1,17 @@
+parameters:
+    level: 3
+    paths:
+        - lib
+    ignoreErrors:
+        # Making classes final as suggested would be a BC-break
+        -
+            message: '~Unsafe usage of new static\(\)\.~'
+            paths:
+                - 'lib/Doctrine/Common/Collections/ArrayCollection.php'
+                - 'lib/Doctrine/Common/Collections/Criteria.php'
+        -
+            message: '~Array \(array\<TKey, T\>\) does not accept key int\.~'
+            path: 'lib/Doctrine/Common/Collections/ArrayCollection.php'
+        -
+            message: '~Method Doctrine\\Common\\Collections\\ArrayCollection\:\:map\(\) should return Doctrine\\Common\\Collections\\ArrayCollection\<TKey, U\> but returns Doctrine\\Common\\Collections\\ArrayCollection\<TKey, T\>\.~'
+            path: 'lib/Doctrine/Common/Collections/ArrayCollection.php'
```

Comment:
@weirdan @someniatko if maybe you could have a look at these errors I ignored, it would be great.

Reply:
Oh, thank you for invitation, it was a bit of unexpected! I guess weirdan is right, as long as `createFrom()` function is a static one, it should be "templated" by its own types instead of TKey and T of the class.

Reply:
Yup, done! Thanks for chiming in!

Reply:
> as long as createFrom() function is a static one, it should be "templated" by its own types instead of TKey and T of the class.

It's not static, but it creates a new, unrelated collection instance. Thus it should not be constrained by the types of the instance it was called on.

Reply:
Oh, I was about to write "behaves as a static function", but had actually written not what i wanted 😌